### ISSUE TITLE
Rephrase in terms of completion values

### ISSUE THREAD

**Issue by domenic** [domenic/promises-unwrapping#6]
They are a bit more awkward than our current "if this call throws an exception," but give some extra precision.

**Comment by erights** [comment]
Extra precision how?

**Comment by domenic** [comment]
Yeah, maybe not that much extra precision.

Notably, in doing #13, I decided to cave and use the phrasing "For each value `nextValue` of `iterable`" for iteration, instead of using the ES spec's `IteratorNext`/`IteratorComplete`/`IteratorValue` and completion value low-level stuff.

I think I've reached an uneasy medium between direct ES spec formalism and something more loose like the DOM or code-as-specification. I'll leave it. If implementers find the current formalism confusing, we can revisit moving it more in the direction of the formal ES spec.
